docs/openssl.rst

```diff
@@ -44,6 +44,15 @@ control.
 
         This will activate the default OpenSSL CSPRNG.
 
+.. _legacy-provider:
+
+Legacy provider in OpenSSL 3.x
+------------------------------
+As of ``cryptography`` 39.0.0 we support OpenSSL 3.x with the legacy provider
+disabled if the user sets the environment variable
+``CRYPTOGRAPHY_OPENSSL_NO_LEGACY``. In this configuration some legacy
+cryptographic algorithms are not available.
```

Comment:
This is very confusingly worded. I suggest using the `versionadded:: 39.0.0` to express the new version, and then describing the functionality in straight prose.